executor/src/witgen/machines/block_machine.rs

```diff
@@ -160,13 +163,39 @@ impl<'a, T: FieldElement> Machine<'a, T> for BlockMachine<'a, T> {
         self.parts.connections.keys().copied().collect()
     }
 
+    fn can_process_call_fully(
+        &mut self,
+        can_process: impl CanProcessCall<T>,
+        identity_id: u64,
+        known_arguments: &BitVec,
+        _range_constraints: &[RangeConstraint<T>],
+    ) -> Option<Vec<RangeConstraint<T>>> {
+        // We do not use the input range constraints because then we would need
+        // to generate new code depending on the range constraints as well.
+        self.function_cache
+            .compile_cached(can_process, identity_id, known_arguments)
```

Comment:
This also calls `compile_effects`, which could take a long time, because it also compiles Rust.

I suppose that's fine because we'll need the Rust function anyway?

Reply:
I mean yes, we kind of need to fully complete it. In the future we could do the actual compilation in the background and run the interpreter until it is finished. In some cases, deriving the code could also take much longer than the compilation....

This is for future georg and chris to solve.

Reply:
I'm just asking because if we just needed to answer whether we can process the call, we could stop after the effects are generated. But I think in all cases where we answer yes, we'd also actually use the code, so the compilation from Rust is not for nothing.